manager/integration/tests/common.py

```diff
@@ -1506,7 +1506,7 @@ def write_volume_random_data(volume, position={}):
 
 def check_device_data(dev, data, check_checksum=True):
     r_data = dev_read(dev, data['pos'], data['len'])
-    assert r_data == data['content']
+    assert r_data == bytes(data['content'], encoding='utf8')
```

Comment:
Do we need `encoding`? Random data is obviously not utf8.

Reply:
Type of `data['contents']` is an object of string class, and type of `r_data` is an object of bytes class
encoding here is needed to convert string to bytes and compare.

Reply:
Do we need to specify encoding `utf8`? By reading this code, one might think the data is encoded in utf8. But it's just raw data in alphabets. The encoding shouldn't matter so it's confusing.

Reply:
When encoding is not used, assert is always `False` even if the data matched.

for example if `data['content']` is a string like this `"ABC....."`
`r_data` is `b'ABC.....'`

Reply:
Looks like Python 3 enforced choosing the encoding for string:

>If it is a string, you must also give the encoding (and optionally, errors) parameters; bytearray() then converts the string to bytes using str.encode().